src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -810,12 +807,28 @@ void InterpreterSelectQuery::executeImpl(QueryPlan & query_plan, const BlockInpu
     bool intermediate_stage = false;
     bool to_aggregation_stage = false;
     bool from_aggregation_stage = false;
+    const bool filter_in_prewhere = (
+        (settings.optimize_move_to_prewhere || expressions.prewhere_info) &&
```

Comment:
Also comment will be helpful why this condition is used:
* `settings.optimize_move_to_prewhere` - it is allowed to move something to prewhere, so we do it for row level actions
* `expressions.prewhere_info` - we already have prewhere and must execute row level filter before it
* `!input && !input_pipe && storage && storage->supportsPrewhere()` - check that prewhere can be used at all

Reply:
Done.